src/tool_operate.c

```diff
@@ -654,11 +654,13 @@ static CURLcode post_per_transfer(struct GlobalConfig *global,
     }
     if(result && config->rm_partial) {
       struct_stat st;
-      stat(outs->filename, &st);
-      if(S_ISREG(st.st_mode)) {
+      if(!stat(outs->filename, &st) &&
+         S_ISREG(st.st_mode)) {
         notef(global, "Removing output file: %s", outs->filename);
         unlink(outs->filename);
       }
+      else
+        warnf(global, "Failed removing: %s", outs->filename);
```

Comment:
Technically, this isn't going to be displayed when removal fails, but only when the removal isn't even attempted.

Reply:
Well, the conditions prevented it from trying, so in that sense it failed to do it.

Any suggestion on a better wording?

Reply:
Maybe "Skipping removal; not a regular file: %s"